### ISSUE TITLE
Find All References does not count usages from the Implicit Range support

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#41310]
**Version Used**: 
16.4.3

**Steps to Reproduce**:
```csharp
public class T
{
    public int Length => 42;
    public int Slice(int start, int length) => start + length;
    public static void M()
    {
        var t = new T();
        var o = t[0..^1];
    }
}
```
Run Find All References on `Length` or `Slice` or look at CodeLense

**Expected Behavior**:
Both `Length` and `Slice` should have one reference from using the implicit range in `M()`.

**Actual Behavior**:
The implicit usage is not reported.

![image](https://user-images.githubusercontent.com/1947968/73452986-60093800-436b-11ea-8a88-21566000a8c5.png)

**Comment by jcouv** [comment]
FYI @agocke 

For what it's worth, here's how this worked for `Deconstruct` methods:
https://github.com/dotnet/roslyn/pull/22934
https://github.com/dotnet/roslyn/pull/24223 (fixup)

**Comment by allisonchou** [comment]
This issue is currently blocked by #44202.

**Comment by sharwell** [comment]
@allisonchou Can you make sure the tests for this include the cases mentioned in #51443?

**Comment by CyrusNajmabadi** [comment]
We need a compiler-api for this (or the semantic model returning the right info here).
